File: C++/021_Merge Two Sorted Lists.cpp
class Solution {
public:
    ListNode *mergeTwoLists(ListNode *l1, ListNode *l2) {
        ListNode *head = new ListNode(-1);
        ListNode *now = head;
        while (l1 != NULL && l2 != NULL) {
            if (l1->val < l2->val) {
                now->next = l1;
                l1 = l1->next;
            }
            else {
                now->next = l2;
                l2 = l2->next;
            }
            now = now->next;
        }
        while (l1 != NULL) {
            now->next = l1;
            l1 = l1->next;
            now = now->next;
        }
        while (l2 != NULL) {
            now->next = l2;
            l2 = l2->next;
            now = now->next;
        }
        now = head->next;
        delete head;
        return now;
    }
};

File: Python/021_Merge Two Sorted Lists.py
class Solution(object):
    def mergeTwoLists(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        head, pre = None, None
        while l1 is not None and l2 is not None:
            if l1.val <= l2.val:
                if head is None:
                    head = l1
                else:
                    pre.next = l1
                pre = l1
                l1 = l1.next
            else:
                if head is None:
                    head = l2
                else:
                    pre.next = l2
                pre = l2
                l2 = l2.next
        while l1 is not None:
            if head is None:
                head = l1
            else:
                pre.next = l1
            pre = l1
            l1 = l1.next
        while l2 is not None:
            if head is None:
                head = l2
            else:
                pre.next = l2
            pre = l2
            l2 = l2.next
        return head